pokemonduel/misc.py

```diff
@@ -429,6 +429,9 @@ def apply_status(self, status, battle, *, attacker=None, move=None, turns=None,
             self.current = status
             self.sleep_timer.set_turns(turns)
             msg += f"{self.pokemon.name} fell asleep{source}!\n"
+            if self.pokemon.last_move.effect == 38:

```

Comment:
Similar to #76, the more correct way to check for the move is the use the `move` argument to `apply_status`. That ensures the move is being used to apply the status, rather than just hoping the last move is the correct one.